dart/common/LocalResourceRetriever.h

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2015, Georgia Tech Research Corporation
+ * All rights reserved.
+ *
+ * Author(s): Michael Koval <mkoval@cs.cmu.edu>
+ *
+ * Georgia Tech Graphics Lab and Humanoid Robotics Lab
+ *
+ * Directed by Prof. C. Karen Liu and Prof. Mike Stilman
+ * <karenliu@cc.gatech.edu> <mstilman@cc.gatech.edu>
+ *
+ * This file is provided under the following "BSD-style" License:
+ *   Redistribution and use in source and binary forms, with or
+ *   without modification, are permitted provided that the following
+ *   conditions are met:
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
+ *   CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ *   INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ *   MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ *   DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ *   CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ *   SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ *   LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+ *   USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ *   AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *   LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *   ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *   POSSIBILITY OF SUCH DAMAGE.
+ */
+#ifndef DART_COMMON_LOCALRESOURCERETRIEVER_H_
+#define DART_COMMON_LOCALRESOURCERETRIEVER_H_
+#include "dart/common/ResourceRetriever.h"
+
+namespace dart {
+namespace common {
+
+/// LocalResourceRetriever provides access to local resources specified by
+/// file:// URIs by wrapping the standard C and C++ file manipulation routines.
+class LocalResourceRetriever : public virtual ResourceRetriever
+{
+public:
+  virtual ~LocalResourceRetriever() = default;
+
+  // Documentation inherited.
+  bool exists(const std::string& _uri) override;
+
+  // Documentation inherited.
+  ResourcePtr retrieve(const std::string& _uri) override;
+};
+
+typedef std::shared_ptr<LocalResourceRetriever> LocalResourceRetrieverPtr;
```

Comment:
How about using `using` rather than `typedef` for type alias as:
```cpp
using LocalResourceRetrieverPtr = std::shared_ptr<LocalResourceRetriever>;
```

They are same but [it seems `using` is more preferred](http://stackoverflow.com/questions/24916435/why-do-type-aliases-in-c-use-using-instead-of-typedef-in-their-syntax).